tests/test_handler.py

```diff
@@ -284,3 +287,130 @@ async def test_stats(mockserver):
             value = scrapy_stats.get_value(stat)
             assert isinstance(value, float)
             assert value > 0.0
+
+
+@ensureDeferred
+@pytest.mark.parametrize(
+    "settings,enabled",
+    [
+        ({}, False),
+        ({"ZYTE_API_LOG_REQUESTS": False}, False),
+        ({"ZYTE_API_LOG_REQUESTS": True}, True),
+    ],
+)
+async def test_log_request_toggle(
+    settings: Dict[str, Any],
+    enabled: bool,
+    mockserver,
+):
+    async with make_handler(settings, mockserver.urljoin("/")) as handler:
+        meta = {"zyte_api": {"foo": "bar"}}
+        request = Request("https://example.com", meta=meta)
+        with mock.patch("scrapy_zyte_api.handler.logger") as logger:
+            await handler.download_request(request, None)
+        if enabled:
+            logger.debug.assert_called()
+        else:
+            logger.debug.assert_not_called()
+
+
+@ensureDeferred
+@pytest.mark.skipif(sys.version_info < (3, 8), reason="unittest.mock.AsyncMock")
+@pytest.mark.parametrize(
+    "settings,short_str,long_str,truncated_str",
+    [
+        ({}, "a" * 64, "a" * 65, "a" * 63 + "…"),
+        ({"ZYTE_API_LOG_REQUESTS_TRUNCATE": 0}, "a" * 64, "a" * 65, "a" * 65),
+        ({"ZYTE_API_LOG_REQUESTS_TRUNCATE": 1}, "a", "aa", "…"),
+        ({"ZYTE_API_LOG_REQUESTS_TRUNCATE": 2}, "aa", "aaa", "a…"),
```

Comment:
In the implementation, I went with an Unicode ellipsis (…) thinking Windows now supports UTF-8 terminal output. But re-reading about it, it seems users actually need to configure things to get UTF-8 terminal output, it is not enabled by default.

The problem with the ASCII alternative (...) is that it is 3-characters long. Shall we set a minimum value for truncation of 3 instead of 1, and disallow 1 and 2 as values for the setting?

Reply:
>  it seems users actually need to configure things to get UTF-8 terminal output, it is not enabled by default.
Could you send a link to a doc/article regarding this?

I'm testing it in my Windows machine and "…" currently works without any problems.